pipeline/outputs/s3.md

```diff
@@ -25,9 +25,9 @@ Records are stored in files in S3 as newline delimited JSON.
 | upload\_chunk\_size | The size of each 'part' for multipart uploads. Max: 50M | 5,242,880 bytes |
 | upload\_timeout | Whenever this amount of time has elapsed, Fluent Bit will complete an upload and create a new file in S3. For example, set this value to 60m and you will get a new file every hour. | 10m |
 | store\_dir | Directory to locally buffer data before sending. When multipart uploads are used, data will only be buffered until the `upload_chunk_size` is reached. | /tmp/fluent-bit/s3 |
-| s3\_key\_format | Format string for keys in S3. This option supports strftime time formatters and a syntax for selecting parts of the Fluent log tag using a syntax inspired by the rewrite\_tag filter. Add $TAG in the format string to insert the full log tag; add $TAG\[0\] to insert the first part of the tag in the s3 key. The tag is split into “parts” using the characters specified with the `s3_key_format_tag_delimiters` option. See the in depth examples and tutorial in the documentation. | /fluent-bit-logs/$TAG/%Y/%m/%d/%H/%M/%S |
+| s3\_key\_format | Format string for keys in S3. This option supports an UUID, strftime time formatters, a syntax for selecting parts of the Fluent log tag using a syntax inspired by the rewrite\_tag filter and key suffix. Add $UUID in the format string to insert a random string. Add $TAG in the format string to insert the full log tag; add $TAG\[0\] to insert the first part of the tag in the s3 key. The tag is split into “parts” using the characters specified with the `s3_key_format_tag_delimiters` option. Add extension directly after the last piece of the format string to insert key suffix. See the in depth examples and tutorial in the documentation. | /fluent-bit-logs/$TAG/%Y/%m/%d/%H/%M/%S/$UUID.gz |
```

Comment:
Nit:

> an UUID

I'd write "a UUID". The rule I was taught in school is that:
* "a" if the word is pronounced starting with a consonant
* "an" if its pronounced starting with a vowel

"U" is a vowel but its pronounced like "you" and "y" is a consonant in that case. Hence "a UUID".

Looks like others use my version, see the title here for example: https://rclayton.silvrback.com/do-you-really-need-a-uuid-guid

Reply:
Non nit:

> | /fluent-bit-logs/$TAG/%Y/%m/%d/%H/%M/%S/$UUID.gz

You can't change the default value. The last column is used to show the default, and that didn't change.

Reply:
its a good idea to show an example with ".gz" just don't put it where the default should go. Put it in the description or in the tutorial.

Also i think you need to note that if they don't include $UUID, and they use put object mode, then we will append a UUID at the end. And then state that if they want to customize the extension they must specify UUID. I think all this info needs to be in the description here so that people can't miss it.

Reply:
Thanks for the suggestion